packages/grid_client/src/clients/tf-grid/contracts.ts

```diff
@@ -282,7 +282,11 @@ class TFContracts extends Contracts {
    * @returns {Promise<Consumption>} A promise resolving to the consumption details,
    * including the amount billed and the discount received.
    */
-  async getConsumption(options: GetConsumptionOptions): Promise<Consumption> {
+  async getConsumption(
+    options: GetConsumptionOptions,
+    contract: Contract,
+    proxy: GridProxyClient,
```

Comment:
We may have a new function called, for example, `getConsumptionWithEstimation`, 
As these changes will break the current usage of this function in other places, and maybe in other clients